### PR TITLE
Bump FIRRTL; Bump FC -> Reenables BOOM

### PR DESCRIPTION
**Pull request by davidbiancolin**
This gets all the assembly tests passing on BOOM in MIDAS-level simulation, though it still locks up just after mounting the rootfs. This can be reproduced trivially in target-level simulation (in FireChip). 

TODOs:

- [x] BOOM doc updates (David)
- [x] run-all.sh fixes (Howie)
- [x] regenerate prebuilt tools (rc-bump-may branch)
- [x] reproduce multi-16x-large results (David)
- [x] AT MERGE TIME: merge pre-built tools repo branch (how does this affect people trying to use the old version? they should get a hash mismatch and then it'll install tools the normal way)
- [ ] AT MERGE TIME: update changelog release date, create release.

### CODE REVIEW

**Comment by davidbiancolin** [comment]
Ping tests check out. Regenerating the bitstreams for the remaining standard designs. 

@zhemao. You need to fix run-all.sh on this branch though.

File: sim/src/main/scala/TargetConfigs.scala [hunk 1]
```diff
@@ -5,22 +5,24 @@ import freechips.rocketchip.tile._
...
-//import boom.system.BoomTilesKey
+import boom.system.BoomTilesKey
 import testchipip.{WithBlockDevice, BlockDeviceKey, BlockDeviceConfig}
 import sifive.blocks.devices.uart.{PeripheryUARTKey, UARTParams}
 import icenet._
 
 class WithBootROM extends Config((site, here, up) => {
   case BootROMParams => BootROMParams(
-    contentFileName = s"./target-rtl/firechip/bootrom/bootrom.rv${site(XLen)}.img")
+    contentFileName = s"./target-rtl/firechip/testchipip/bootrom/bootrom.rv${site(XLen)}.img")
```
Comment (sagark) [inline sim/src/main/scala/TargetConfigs.scala]: Is the `firesim/sim/bootrom` directory still necessary? If not, we should remove it.

**Comment by sagark** [comment]
- `firesim-riscv-tools-prebuilt` needs to be updated to the riscv-tools in this version. The updated version should sit on a branch (not master) until this PR is merged.

**Comment by sagark** [comment]
- Need docs for BOOM (that also reference the issue about it not completing linux boot)

What's the status of running sample experiments on this version?

**Comment by sagark** [comment]
@zhemao does this include updates to the NIC RTL/Linux driver?

**Comment by zhemao** [comment]
Updates to NIC RTL aren't included in this PR. I've added changes to get run-bw-test.sh working again.

**Comment by colinschmidt** [comment]
Status?

**Comment by zhemao** [comment]
@colinschmidt We're waiting for David to finish is action items and sign off on them.

File: sim/src/main/scala/TargetConfigs.scala (hunk 1 above)
Comment (sagark) [reply sim/src/main/scala/TargetConfigs.scala]: @zhemao

**Comment by davidbiancolin** [comment]
This is ready for review. @sagark

**Comment by sagark** [comment]
@davidbiancolin can we add a note and a link to the issue about BOOM booting Linux in the docs?

**Comment by sagark** [comment]
@zhemao can you address the bootrom comment above?

**Comment by sagark** [comment]
@davidbiancolin the pre-generated image AGFIs included in this branch are public right? (not just open to ucb-bar folks)

**Comment by sagark** [comment]
@davidbiancolin @zhemao can you guys confirm that CHANGELOG.md reflects all changes? https://github.com/firesim/firesim/blob/rc-bump-may/CHANGELOG.md#120---2018-06-tbd

**Comment by sagark** [comment]
@zhemao @davidbiancolin one more thing, are we all sync'd up with masters of other repos we control (as outlined in CONTRIBUTING.md)? 

What we want is that submodules that we control have master branches that match what's currently submoduled in firesim master.

**Comment by davidbiancolin** [comment]
AGFIs are public yes. I figured with the exception of MIDAS, we can merge the branches into master after this is merged in.

**Comment by zhemao** [comment]
The network-benchmarks changes need to be merged in, but otherwise I think we're good. I created a PR for network-benchmarks.

**Comment by davidbiancolin** [comment]
Changelog looks good. Added the note about BOOM support being in beta.

**Pull request merged**